python/sbp/client/framer.py

```diff
@@ -101,29 +104,33 @@ def _receive(self):
     # empty input
     preamble = self._read(1)
     if not preamble:
+      print 'not preample'
       return None
     elif ord(preamble) != SBP_PREAMBLE:
       if self._verbose:
         print "Host Side Unhandled byte: 0x%02x" % ord(preamble)
       return None
+
     # hdr
     hdr = self._readall(5)
-    msg_crc = crc16(hdr)
-    msg_type, sender, msg_len = struct.unpack("<HHB", hdr)
+    msg_type, sender, msg_len = np.ndarray(1, 'u2, u2, u1', hdr)[0]
+
     # data
     data = self._readall(msg_len)
-    msg_crc = crc16(data, msg_crc)
+    msg_crc = crc16(hdr + data)
+
     # crc
     crc = self._readall(2)
-    crc, = struct.unpack("<H", crc)
+    crc = np.ndarray(1, 'u2', crc)[0]
     if crc != msg_crc:
       print "crc mismatch: 0x%04X 0x%04X" % (msg_crc, crc)
       return None
+
     msg = SBP(msg_type, sender, msg_len, data, crc)
     try:
       msg = self._dispatch(msg)
     except:
-      pass
+      print 'error'
```

Comment:
Can we make this error more informative?  
"Error dispatching sbp msg of type: "